modules/tinymce/src/plugins/preview/test/ts/browser/PreviewSanityTest.ts

```diff
@@ -1,46 +1,39 @@
-import { TinyApis, TinyDom, TinyLoader, TinyUi } from '@ephox/mcagar';
+import { Keyboard, Keys, UiFinder, Waiter } from '@ephox/agar';
+import { describe, it } from '@ephox/bedrock-client';
+import { TinyDom, TinyHooks, TinyUiActions } from '@ephox/mcagar';
 import { SugarElement } from '@ephox/sugar';
 
-import PreviewPlugin from 'tinymce/plugins/preview/Plugin';
-import SilverTheme from 'tinymce/themes/silver/Theme';
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/preview/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
 
-UnitTest.asynctest('browser.tinymce.plugins.preview.PreviewSanityTest', (success, failure) => {
-
-  PreviewPlugin();
-  SilverTheme();
+describe('browser.tinymce.plugins.preview.PreviewSanityTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    plugins: 'preview',
+    toolbar: 'preview',
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Theme, Plugin ]);
 
   const dialogSelector = 'div[role="dialog"]';
   const docBody = SugarElement.fromDom(document.body);
-  const doc = TinyDom.fromDom(document);
-
-  TinyLoader.setupLight((editor, onSuccess, onFailure) => {
-    const tinyApis = TinyApis(editor);
-    const tinyUi = TinyUi(editor);
-
-    const sOpenDialog = () => GeneralSteps.sequence(Logger.ts('Open dialog and wait for it to be visible', [
-      tinyUi.sClickOnToolbar('click on preview toolbar', 'button'),
-      tinyUi.sWaitForPopup('wait for preview popup', '[role="dialog"] iframe')
-    ]));
 
-    Pipeline.async({},
-      Log.steps('TBA', 'Preview: Set content, open dialog, click Close to close dialog. Open dialog, press escape and assert dialog closes', [
-        tinyApis.sSetContent('<strong>a</strong>'),
-
-        sOpenDialog(),
-        tinyUi.sClickOnUi('Click on Close button', '.tox-button:not(.tox-button--secondary)'),
-        Waiter.sTryUntil('Dialog should close', UiFinder.sNotExists(docBody, dialogSelector)),
-
-        sOpenDialog(),
-        Keyboard.sKeydown(doc, Keys.escape(), { }),
-        Waiter.sTryUntil('Dialog should close on esc', UiFinder.sNotExists(docBody, dialogSelector))
-      ])
-      , onSuccess, onFailure);
-  }, {
-    theme: 'silver',
-    plugins: 'preview',
-    toolbar: 'preview',
-    base_url: '/project/tinymce/js/tinymce'
-  }, success, failure);
+  const openDialog = async (editor: Editor) => {
+    TinyUiActions.clickOnToolbar(editor, 'button');
+    await TinyUiActions.pWaitForPopup(editor, '[role="dialog"] iframe');
+  };
+
+  it('TBA: Set content, open dialog, click Close to close dialog. Open dialog, press escape and assert dialog closes', async () => {
+    const editor = hook.editor();
+    editor.setContent('<strong>a</strong>');
+
+    await openDialog(editor);
+    // Close dialog with button
+    TinyUiActions.clickOnUi(editor, '.tox-button:not(.tox-button--secondary)');
+    await Waiter.pTryUntil('Dialog should close', () => UiFinder.notExists(docBody, dialogSelector));
```

Comment:
We really need to add this type of thing to `UiFinder` one day, as we have one for exists, visible and hidden, but not for "not exists".

Reply:
Agree